### ISSUE TITLE
Cleaning no longer removes app project assembly and pdb files from IntermediateOutputPath on macOS

### ISSUE THREAD

**Issue by pjcollins** [xamarin/xamarin-android#1824]
### Steps to Reproduce
https://github.com/xamarin/xamarin-forms-samples/tree/master/WorkingWithStyles/Android

1. Clone or download the sample above.
2. `nuget restore xamarin-forms-samples/WorkingWithStyles/WorkingWithStyles.sln`
3. `msbuild /t:SignAndroidPackage xamarin-forms-samples/WorkingWithStyles/Android/WorkingWithStyles.Android.csproj`
4.  `msbuild /t:Clean xamarin-forms-samples/WorkingWithStyles/Android/WorkingWithStyles.Android.csproj`

### Expected Behavior
Cleaning a project should empty the IntermediateOutputPath.

### Actual Behavior
The following .dll and .pdb files are left behind after cleaning. This behavior is a regression in recent d15-8 builds.
* xamarin-forms-samples/WorkingWithStyles/Android/obj/Debug/WorkingWithStyles.Android.dll
* xamarin-forms-samples/WorkingWithStyles/Android/obj/Debug/WorkingWithStyles.Android.pdb

### Version Information
XA 15.8 (8.4.0.1)

### Log File
Clean output:
https://gist.github.com/pjcollins/2965f5132354f9ce24a5aabe443d5cb9

> VS bug [#633699](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/633699)

**Comment by dellis1972** [comment]
@pjcollins any chance we can check if a normal .net console app has the same problem? We haven't changed any code in the `Clean` targets in ages. 

The only thing it might be is fallout from the `IntermediateOutputPath` changes made by the vs team [1]

[1] https://github.com/xamarin/xamarin-android/commit/c6e9fc95f5026ba0e1be5da51f59256c9a706ed4

**Comment by jonathanpeppers** [comment]
Ok, so if I look at `/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/4.5/Microsoft.Common.targets`, it has this for macOS:
```
<PropertyGroup>
    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' == ''">obj\</BaseIntermediateOutputPath>
    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' != '' and !HasTrailingSlash('$(BaseIntermediateOutputPath)')">$(BaseIntermediateOutputPath)\</BaseIntermediateOutput>
</PropertyGroup>
```

It doesn't seem like we are doing the right thing in the original PR: https://github.com/xamarin/xamarin-android/pull/1628

I don't see how it would work on macOS.

**Comment by pjcollins** [comment]
These files are deleted by clean when using the current Alpha channel in VS Mac (XA 8.3.99.12). After upgrading XA builds these files are no longer removed by clean, so I am fairly confident that the change in behavior was caused by a change in XA.

**Comment by dellis1972** [comment]
Hmm. This is weird. Our unit test which checks the Clean target is passing..

**Comment by dellis1972** [comment]
with PR #1826 applied the assemblies seem to be gone. but I have a `netstandard2.0` folder left over with some xaml.g.cs file left over from (it looks like) the Xamarin forms code generator.

**Comment by dellis1972** [comment]
looks like the code generator in XF is not cleaning up after itself

**Comment by jonathanpeppers** [comment]
@dellis1972 that was because they had a typo and were using FilesWrite instead of FileWrites.

They got those targets fixes in the latest 3.0.0.x release.
